crates/red_knot_workspace/src/site_packages.rs

```diff
@@ -84,29 +89,26 @@ fn site_packages_dir_from_sys_prefix(
         // the `pyvenv.cfg` file anyway, in which case we could switch to that method
         // rather than iterating through the whole directory until we find
         // an entry where the last component of the path starts with `python3.`
-        if !path
-            .components()
-            .next_back()
-            .is_some_and(|last_part| last_part.as_str().starts_with("python3."))
-        {
+        let Some(name) = path.file_name() else {
+            continue;
```

Comment:
Since this is a path yielded by a call to read_directory, I don't think it's possible for `path.file_name()` to be `None`. Given that, I find the current code on `main` marginally easier to understand, since this `continue` is essentially dead code. We could also use `path.file_name().expect(...)`, I suppose?

Reply:
(But using `.file_name()` instead of `.components.next_back()` is definitely an improvement, of course :-)